diagnostic_feedback/tests/unit/test_student_view.py

```diff
@@ -38,56 +38,59 @@ def load_json_resource(self, path):
 
     def setUp(self):
         self._block = self.make_block()
-        self._daignostic_answer = json.loads(self.load_json_resource('data/answer_diagnostic_test_data.json'))
+        self._diagnostic_answer = json.loads(self.load_json_resource('data/answer_diagnostic_test_data.json'))
         self._buzzfeed_answer = json.loads(self.load_json_resource('data/answer_buzzfeed_test_data.json'))
 
     def test_diagnostic_answer(self):
 
         # add diagnostic quiz
         res = self.save_wizard_step1(self._block.DIAGNOSTIC_QUIZ_VALUE)
-        assert_equals(res['success'], True)
+        assert_true(res['success'])
 
         # add ranges for diagnostic quiz
         res = self.save_diagnostic_step2()
-        assert_equals(res['success'], True)
+        assert_true(res['success'])
 
         # add Questions for diagnostic Quiz
         res = self.save_diagnostic_step3()
-        assert_equals(res['success'], True)
+        assert_true(res['success'])
 
         for question_data in self._block.questions:
-            for _type, data in self._daignostic_answer.items():
+            for _type, data in self._diagnostic_answer.items():
                 if _type == 'missing_id':
                     data['question_id'] = ''
                 else:
                     data['question_id'] = question_data["id"]
                 json_data = json.dumps(data)
                 res = json.loads(self._block.handle('save_choice', self.make_request(json_data)).body)
                 if _type == 'missing_choice':
-                    assert_equals(res['success'], False)
+                    assert_false(res['success'])
                 elif _type == 'missing_id':
-                    assert_equals(res['success'], False)
+                    assert_false(res['success'])
                 elif _type == 'missing_step':
-                    assert_equals(res['success'], False)
-                if _type == 'valid_data':
+                    assert_false(res['success'])
+                elif _type == 'valid_data':
                     if data['isLast'] == "True":
-                        assert_equals(res['success'], True)
+                        assert_true(res['success'])
                     else:
-                        assert_equals(res['success'], True)
+                        assert_true(res['success'])
```

Comment:
Nit:  It looks like this has the same code on the if and else branches.  (LL73 - 76).  You can do away with the conditional